Aikido.Zen.Core/Models/Ip/Blocklist.cs

```diff
@@ -13,15 +13,17 @@ namespace Aikido.Zen.Core.Models.Ip
     public class BlockList
     {
         // The state of our blocklist needs to be thread-safe, as incoming ASP requests can be multithreaded, and the agent, which runs on a background thread, can also update/access the state.
-        private IPRange _blockedSubnets = new IPRange();
-        private ConcurrentDictionary<string, IPRange> _allowedSubnets = new ConcurrentDictionary<string, IPRange>();
+        private IPRange _blockedIps = new IPRange();
+        private IPRange _allowedIps = new IPRange();
+        private IPRange _bypassedIps = new IPRange();
+        private ConcurrentDictionary<string, IPRange> _allowedIpsPerEndpoint = new ConcurrentDictionary<string, IPRange>();
         private readonly ReaderWriterLockSlim _lock = new ReaderWriterLockSlim();
```

Comment:
This lock is used for _blockedIps, _allowedIps and _byppassedIps if I understand it correctly? Let's comment for what the lock is used.